File: src/packet/payload/lobby_info.rs
use bytemuck::{Pod, Zeroable};
use serde::Serialize;

use crate::packet::header;

#[repr(C, packed)]
#[derive(Clone, Copy, Debug, Zeroable, Pod, Serialize)]
pub struct LobbyInfo {
    pub m_ai_controlled: u8,
    pub m_team_id: u8,
    pub m_nationality: u8,
    pub m_platform: u8,
    pub m_name: [u8; 32],
    pub m_car_number: u8,
    pub m_your_telemetry: u8,
    pub m_show_online_names: u8,
    pub m_tech_level: u16,
    pub m_ready_status: u8,
}

#[repr(C, packed)]
#[derive(Clone, Copy, Debug, Zeroable, Pod, Serialize)]
pub struct PacketLobbyInfo {
    pub m_header: header::PacketHeader,
    pub m_num_players: u8,
    pub m_lobby_players: [LobbyInfo; 22],
}

File: consumer/src/main/scala/packet/payload/LobbyInfo.scala
package packet.payload

import packet.Header

case class LobbyInfo(
    m_ai_controlled: Int,
    m_team_id: Int,
    m_nationality: Int,
    m_platform: Int,
    m_name: Array[Int],
    m_car_number: Int,
    m_your_telemetry: Int,
    m_show_online_names: Int,
    m_tech_level: Int,
    m_ready_status: Int
)

case class PacketLobbyInfo(
    m_header: Header,
    m_num_players: Int,
    m_lobby_players: Array[LobbyInfo]
)
